pkg/validation/internal/annotations.go

```diff
@@ -0,0 +1,65 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/validation/errors"
+)
+
+// CaseSensitiveAnnotationKeySet is a set of annotation keys that are case sensitive
+// and can be used for validation purposes. This may not be an exhaustive list.
+var CaseSensitiveAnnotationKeySet = map[string]struct{}{
+
+	v1.OperatorGroupAnnotationKey:             {},
+	v1.OperatorGroupNamespaceAnnotationKey:    {},
+	v1.OperatorGroupTargetsAnnotationKey:      {},
+	v1.OperatorGroupProvidedAPIsAnnotationKey: {},
+	"olm.skipRange":                           {},
+
+	// below are duplicate values of known keys
+
+	// v1alpha1.OperatorGroupNamespaceAnnotationKey:    {},
+	// v1alpha2.OperatorGroupAnnotationKey:             {},
+	// v1alpha2.OperatorGroupNamespaceAnnotationKey:    {},
+	// v1alpha2.OperatorGroupTargetsAnnotationKey:      {},
+	// v1alpha2.OperatorGroupProvidedAPIsAnnotationKey: {},
+}
+
+/*
+ValidateAnnotationNames will check annotation keys to ensure they are using
+proper case. Uses CaseSensitiveAnnotationKeySet as a source for keys
+which are known to be case sensitive. This function can be used anywhere
+annotations need to be checked for case sensitivity.
+
+Arguments
+
+• annotations: annotations map usually obtained from ObjectMeta.GetAnnotations()
+
+• value: is the field or file that caused an error or warning
+
+Returns
+
+Any errors that may have been detected with the annotation keys provided
+*/
+func ValidateAnnotationNames(annotations map[string]string, value interface{}) (errs []errors.Error) {
+	// for every annotation provided
+	for annotationKey := range annotations {
+		// loop through all well known keys
+		for knownCaseSensitiveKey := range CaseSensitiveAnnotationKeySet {
```

Comment:
You create `CaseSensitiveAnnotationKeySet` as a map. Why don't you use the key look up to check if the key exists or not? It is faster. No need to loop through the entire map like this.

Reply:
On the second look, I can see why you don't use hash look-up because of the case-sensitive nature. See my comment on the map variable.

Reply:
I plan to reimplement based on your other comment.